corehq/motech/openmrs/tests/test_repeaters.py

```diff
@@ -172,7 +193,11 @@ def test_get_case_updates_for_followup(self):
                     },
                     created=False,
                     closed=False,
-                    extra_fields={},
+                    extra_fields={
+                        'name': 'Elsa',
+                        'estado_tarv': '1',
+                        'tb': '0',
+                    },
```

Comment:
not sure how useful but do we want to add a case where its a case property and within extra field as well?

Reply:
Yes. "name", "estado_tarv" (which, btw, is Portuguese for "currently on anti-retrovirals") and "tb" are all case properties, set in the mock that decorates the class on lines 136-138. The code we're testing is run when they are passed to `get_relevant_case_updates_from_form_json()` on line 185. This result here, lines 196-200, show that the values were correctly fetched from the case.

:thinking: Maybe we should test that passing in an extra field that the case doesn't have a property for will return None.

Reply:
Test extended to include an extra field that isn't a case property.

Reply:
:+1: Doctest picks up where this test leaves off.

Reply:
just out of curiosity, when does this doc test run?

Reply:
It gets run by [corehq/motech/tests/test_value_source.py](https://github.com/dimagi/commcare-hq/blob/master/corehq/motech/tests/test_value_source.py#L10)

Reply:
and that would run in travis each time?

Reply:
Yeah. It's just a normal Django SimpleTestCase that runs the doctests. We do it elsewhere too:

```bash
$ git grep -l test_doctests
corehq/apps/app_manager/app_schemas/tests/test_case_properties.py
corehq/apps/app_manager/tests/test_xform_builder.py
corehq/motech/openmrs/tests/test_atom_feed.py
corehq/motech/openmrs/tests/test_finders.py
corehq/motech/openmrs/tests/test_jsonpath.py
corehq/motech/openmrs/tests/test_repeaters.py
corehq/motech/openmrs/tests/test_serializers.py
corehq/motech/tests/test_utils.py
corehq/motech/tests/test_value_source.py
custom/openclinica/tests/test_odm_to_app.py
custom/openclinica/tests/test_utils.py
```

Actually, I don't like what I did in this [Doctest commit](https://github.com/dimagi/commcare-hq/pull/22394/commits/bd3971ca8d8305ea98816d317ae417b523cc5aaa). I believe that doctests are for documentation, not for testing; They are there to show developers what a function or method does, or how it's meant to be used, and one or two tests should be enough. And running your doctests just checks that your documentation is still telling the truth.

Testing all cases, or edge cases, should be done with other kinds of tests. But I made an exception here because these tests are short enough that we could test all three cases in four relatively short statements.

Reply:
> like the same thing being tested as doctest and then as a normal test?

If I understand you correctly, no it only runs once. It's the normal test (i.e. test_value_source.py) that runs the doctest. If it wasn't for the normal test then the doctest wouldn't get run.